orderable/models.py

```diff
@@ -67,6 +67,20 @@ def prev(self):
             sort_order__lt=self.sort_order
         ).order_by('-sort_order').first()
 
+    @staticmethod
+    def _update(qs):
+        """
+        Increment the sort_order in a queryset.
+
+        Handle IntegrityErrors caused by unique constraints.
+        """
+        try:
+            with transaction.atomic():
+                qs.update(sort_order=models.F('sort_order') + 1)
+        except IntegrityError:
+            for obj in qs.order_by('-sort_order'):
+                qs.filter(pk=obj.pk).update(sort_order=models.F('sort_order') + 1)
```

Comment:
This is ugly, but it works...

Unfortunately, Django doesn't have support for [`DEFERRABLE` unique constraints](http://www.postgresql.org/docs/current/static/sql-set-constraints.html). Ideally, we'd be able to set the constraint as `UNIQUE DEFERRABLE INITIALLY IMMEDIATE` and then run the `update` with `DEFERRED` set.